scripts_src/arroyo/aibroc.ssl

```diff
@@ -34,27 +34,38 @@
 #include "../headers/command.h"
 
 procedure start;
+procedure timed_event_p_proc;
 procedure pickup_p_proc;
 procedure destroy_p_proc;
-procedure drop_p_proc;
+//procedure drop_p_proc;
 
 procedure start begin
 end
 
-procedure pickup_p_proc begin
-   if (cur_map_index == MAP_ARROYO_WILDERNESS) then begin
-       set_map_var(MVAR_Current_Xander_Root,map_var(MVAR_Current_Broc_Flower)-1);
+procedure timed_event_p_proc begin
+   variable item;
+   if (obj_is_carrying_obj(dude_obj, self_obj)) then begin
+      rm_obj_from_inven(dude_obj, self_obj);
+      set_map_var(MVAR_Current_Broc_Flower,map_var(MVAR_Current_Broc_Flower)-1);
+      item:=create_object(PID_BROC_FLOWER,0,0);
+      add_obj_to_inven(dude_obj,item);
    end
 end
 
-procedure destroy_p_proc begin
+procedure pickup_p_proc begin
    if (cur_map_index == MAP_ARROYO_WILDERNESS) then begin
-       set_map_var(MVAR_Current_Xander_Root,map_var(MVAR_Current_Broc_Flower)-1);
+      add_timer_event(self_obj,0,1);
```

Comment:
Could be better to move fixed_param into a define.

Reply:
`fixed_param` is never used inside `timed_event_p_proc` though.

Reply:
I understand, still a symbol is better than a magic number.